src/csvcubed/models/sparqlresults.py

```diff
@@ -393,11 +395,66 @@ def _map_qube_component_sparql_result(
     return result
 
 
+def _get_column_components_and_check_for_cube_shape(
```

Comment:
Don't call this `get` since it doesn't return anything. This function *sets* values: the `real_columns_used_in` and `used_by_observed_value_columns` fields; it also sets the `required` property on the component. See if you can come up with a succinct name which describes what it does.